service/history/resource/resourceTest.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package resource
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/uber/cadence/common/metrics"
+	c "github.com/uber/cadence/common/resource"
+	"github.com/uber/cadence/service/history/events"
+)
+
+type (
+	// Test is the test implementation used for testing
+	Test struct {
+		*c.Test
+		eventCache events.Cache
```

Comment:
Let's follow the existing resourceTest Implementation in the common package: export the field and change it's type to `*events.MockCache` so that when doing mock in test we can access it directly and don't need to always perform an additional type assertion.